### PR TITLE
rpcclient: add GetProof and VerifyProof

### LINKED ISSUE #2942: `GetProof` and `VerifyProof` in RpcClient

### PR DESCRIPTION
**Pull request by ZhangTao1596**
Fix #2942

### CODE REVIEW

File: pkg/rpcclient/rpc.go [hunk 1]
```diff
@@ -447,6 +447,31 @@ func (c *Client) GetRawTransactionVerbose(hash util.Uint256) (*result.Transactio
...
+		params = []interface{}{stateroot.StringLE(), historicalContractHash.StringLE(), historicalKey}
+		resp   = &result.ProofWithKey{}
+	)
+	if err := c.performRequest("getproof", params, resp); err != nil {
+		return nil, err
+	}
+	return resp, nil
+}
+
+// VerifyProof returns value by the given stateroot and proof
```
Comment (AnnaShaleva) [inline pkg/rpcclient/rpc.go:463]: Comment should end with a period.

File: pkg/rpcclient/rpc_test.go [hunk 2]
```diff
@@ -831,6 +832,42 @@ var rpcClientTestCases = map[string][]rpcClientTestCase{
 			},
 		},
 	},
+	"getproof": {
```
Comment (AnnaShaleva) [inline pkg/rpcclient/rpc_test.go:835]: Could we add an error case for them as far?

Comment (AnnaShaleva) [inline pkg/rpcclient/rpc_test.go:835]: Could we also add an integrational test to `pkg/services/rpcsrv/client_test.go`, please? We have a special pre-filled testchain, you can get proof related data from it. Check the way how it is done for RPC server tests at the https://github.com/nspcc-dev/neo-go/blob/de7bff9cea3215697c0ecaf851220cd233f05bad/pkg/services/rpcsrv/server_test.go#L2033

Comment (ZhangTao1596) [reply pkg/rpcclient/rpc_test.go:835]: Will do.

Comment (ZhangTao1596) [reply pkg/rpcclient/rpc_test.go:835]: Will do.

Comment (ZhangTao1596) [reply pkg/rpcclient/rpc_test.go:835]: Done.

Comment (ZhangTao1596) [reply pkg/rpcclient/rpc_test.go:835]: Done.

File: pkg/rpcclient/rpc.go (hunk 1 above)
Comment (ZhangTao1596) [reply pkg/rpcclient/rpc.go]: Done.

**Review by AnnaShaleva — COMMENTED** [review]
LGTM!

Could you squash the commits before the merge, please?

**Comment by ZhangTao1596** [comment]
We need fix these golang-lint errors in github workflow. Create another pr?

**Comment by ZhangTao1596** [comment]
> LGTM!
> 
> Could you squash the commits before the merge, please?

Done.

**Comment by AnnaShaleva** [comment]
> We need fix these golang-lint errors in github workflow. Create another pr?

The linter probably has got an update and found a lot of new errors. Your PR's linting is OK, and for the rest of the code I've created #2948, it's a separate task. Let's firstly finish the existing https://github.com/nspcc-dev/neo-go/pull/2892, and after that fix our linter.

**Review by roman-khimov — APPROVED** [review]

**Review by AnnaShaleva — APPROVED** [review]

**Pull request merged**